pta-intelligent-search-metadata-extractor/src/main/java/fi/maanmittauslaitos/pta/search/metadata/ISOMetadataExtractorConfigurationFactory.java

```diff
@@ -143,52 +145,227 @@ public DocumentProcessingConfiguration createMetadataDocumentProcessingConfigura
+				"//gmd:MD_Metadata/gmd:contact/gmd:CI_ResponsibleParty"));
+
+		/*
+			Language data
+		 */
+		// Metadata language
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.LANGUAGE_METADATA,
+				new LanguageXmlCustomExtractor(),
+				"(/csw:GetRecordByIdResponse/*/gmd:language)"));
+
+		//Resource language
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.LANGUAGE_RESOURCE,
+				new LanguageXmlCustomExtractor(),
+				"(//gmd:identificationInfo/*/gmd:language)"));
+
+
+
+		/*
+		 Constraints
+		*/
+		// use limitation
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CONSTRAINT_USE_LIMITATION,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:resourceConstraints/*/gmd:useLimitation/gco:CharacterString/text())"));
+
+		// access limitation
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CONSTRAINT_ACCESS,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:resourceConstraints/*/gmd:accessConstraints/gmd:MD_RestrictionCode/@codeListValue)"));
+
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CONSTRAINT_OTHER,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:resourceConstraints/*/gmd:otherConstraints/gco:CharacterString/text())"));
+
+		/*
+			Spatial data
+		 */
+		final String XPATH_CRS_ROOT = "//gmd:referenceSystemInfo/*/gmd:referenceSystemIdentifier";
+		// CRS code
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CRS_CODE,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(" +
+						XPATH_CRS_ROOT +
+						"/*/gmd:code/gco:CharacterString/text())"));
+
+		// CRS code space
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CRS_CODE_SPACE,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(" +
+						XPATH_CRS_ROOT +
+						"/*/gmd:codeSpace/gco:CharacterString/text())"));
+
+		// CRS version
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CRS_VERSION,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(" +
+						XPATH_CRS_ROOT +
+						"/*/gmd:version/gco:CharacterString/text())"));
+		// Scale data
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.SCALE_DENOMINATOR,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:spatialResolution/*/gmd:equivalentScale/*/gmd:denominator/gco:Integer/text())"));
+
+		// Extent
+		// North / Top
+		final String XPATH_EXTENT_ROOT = "//gmd:identificationInfo/*/*[self::gmd:extent or self::srv:extent]/*/gmd:geographicElement";
+
+
 		// Geographic bounding box
 		extractors.add(createXPathExtractor(
 				ResultMetadataFields.GEOGRAPHIC_BOUNDING_BOX,
 				new GeographicBoundingBoxXmlCustomExtractor(),
 				"//gmd:MD_Metadata/gmd:identificationInfo/*/*[self::gmd:extent or self::srv:extent]/*/gmd:geographicElement/gmd:EX_GeographicBoundingBox"));
 
+		extractors.add(createXPathExtractor(
```

Comment:
Check if the extent values could be read from the above geographic bounding box extractor result. If yes, the four extractors below are not needed

Reply:
Extend values removed